### ISSUE TITLE
"Did not find .cabal file for..." messages with Docker and no ~/.stack

### ISSUE THREAD

**Issue by enolan** [commercialhaskell/stack#2705]
Running Stack with Docker support turned on on a new install produces a bunch of "Did not find .cabal file for" messages. This doesn't happen without Docker turned on.

To reproduce:
- `stack new foo;cd foo`
- add `docker: {enable: true}` to the end of `stack.yaml`
- `rm -rf ~/.stack`
- `stack test`

Logs: https://gist.github.com/enolan/85f21c0287dd0f4cf3332c8fbfa8e2d8
Expected behavior is no warnings, actual behavior is more than 7000 warnings. Version is 1.2.0, happens with both Ubuntu official package and https://www.stackage.org/stack/linux-x86_64 on Travis.

**Comment by mgsloan** [comment]
Which stack version are you using?  The latest version shouldn't do this.

**Comment by enolan** [comment]
It's 1.2.0, I just updated and tested again.

**Comment by sjakobi** [comment]
Could be related to https://github.com/commercialhaskell/stack/issues/2691?!

Although I don't think that the integration test cases are docker-enabled.

**Comment by Tritlo** [comment]
I'm having the same issue. Starts out wit a long scrolling list of these messages, and then continues for all packages as it is building. I'm on version 1.2.0 on OSX El Capitan

**Comment by mgsloan** [comment]
Closing as need-repro:

```
mgsloan@computer:~/fpco/test-stack/issue2705$ mv ~/.stack/ ~/.stack-old
mgsloan@computer:~/fpco/test-stack/issue2705$ stack test
Downloaded lts-8.5 build plan.    
Selected mirror https://s3.amazonaws.com/hackage.fpcomplete.com/                                 
Downloading root                                                                                 
Selected mirror https://s3.amazonaws.com/hackage.fpcomplete.com/                                 
Downloading timestamp                                                                            
Downloading snapshot                                                                             
Downloading mirrors                                                                              
Cannot update index (no local copy)                                                              
Downloading index                                                                                
Updated package list downloaded                                                                  
Populated index cache.                                                                           
Warning: File listed in issue2705.cabal file does not exist: README.md
[1 of 2] Compiling Main             ( /home/mgsloan/.stack/setup-exe-src/setup-mPHDZzAJ.hs, /home/mgsloan/.stack/setup-exe-src/setup-mPHDZzAJ.o )
[2 of 2] Compiling StackSetupShim   ( /home/mgsloan/.stack/setup-exe-src/setup-shim-mPHDZzAJ.hs, /home/mgsloan/.stack/setup-exe-src/setup-shim-mPHDZzAJ.o )
Linking /home/mgsloan/.stack/setup-exe-cache/x86_64-linux-dkda49f7ca9b244180d3cfb1987cbc9743/tmp-Cabal-simple_mPHDZzAJ_1.24.2.0_ghc-8.0.2 ...
issue2705-0.1.0.0: configure (lib + exe + test)
Configuring issue2705-0.1.0.0...
issue2705-0.1.0.0: build (lib + exe + test)
Preprocessing library issue2705-0.1.0.0...
[1 of 1] Compiling Lib              ( src/Lib.hs, .stack-work/dist/x86_64-linux-dkda49f7ca9b244180d3cfb1987cbc9743/Cabal-1.24.2.0/build/Lib.o )
Preprocessing executable 'issue2705-exe' for issue2705-0.1.0.0...
[1 of 1] Compiling Main             ( app/Main.hs, .stack-work/dist/x86_64-linux-dkda49f7ca9b244180d3cfb1987cbc9743/Cabal-1.24.2.0/build/issue2705-exe/issue2705-exe-tmp/Main.o )
Linking .stack-work/dist/x86_64-linux-dkda49f7ca9b244180d3cfb1987cbc9743/Cabal-1.24.2.0/build/issue2705-exe/issue2705-exe ...
Preprocessing test suite 'issue2705-test' for issue2705-0.1.0.0...
[1 of 1] Compiling Main             ( test/Spec.hs, .stack-work/dist/x86_64-linux-dkda49f7ca9b244180d3cfb1987cbc9743/Cabal-1.24.2.0/build/issue2705-test/issue2705-test-tmp/Main.o )
Linking .stack-work/dist/x86_64-linux-dkda49f7ca9b244180d3cfb1987cbc9743/Cabal-1.24.2.0/build/issue2705-test/issue2705-test ...
Warning: File listed in issue2705.cabal file does not exist: README.md
issue2705-0.1.0.0: copy/register
Installing library in
/home/mgsloan/fpco/test-stack/issue2705/.stack-work/install/x86_64-linux-dkda49f7ca9b244180d3cfb1987cbc9743/lts-8.5/8.0.2/lib/x86_64-linux-ghc-8.0.2/issue2705-0.1.0.0-AuYk635UKm79MA9kwgNCye
Installing executable(s) in
/home/mgsloan/fpco/test-stack/issue2705/.stack-work/install/x86_64-linux-dkda49f7ca9b244180d3cfb1987cbc9743/lts-8.5/8.0.2/bin
Registering issue2705-0.1.0.0...
issue2705-0.1.0.0: test (suite: issue2705-test)
             
Progress: 1/2Test suite not yet implemented                                                                                                       
Completed 2 action(s).
```

**Comment by enolan** [comment]
I can't reproduce this anymore either. Probably fixed some time between 1.2.0 and 1.4.0.
